core/ledger/kvledger/kv_ledger.go

```diff
@@ -174,6 +175,77 @@ func newKVLedger(initializer *lgrInitializer) (*kvLedger, error) {
 	return l, nil
 }
 
+func (l *kvLedger) registerStateDBIndexCreatorForChaincodeLifecycleEvents(
+	stateDBIndexCreator cceventmgmt.ChaincodeLifecycleEventListener,
+	deployedChaincodesInfoExtractor ledger.DeployedChaincodeInfoProvider,
+	chaincodesLifecycleEventsProvider ledger.ChaincodeLifecycleEventProvider,
+	legacyChaincodesLifecycleEventsProvider *cceventmgmt.Mgr,
+	bootstrappingFromSnapshot bool,
+) error {
+	if !bootstrappingFromSnapshot {
+		// regular opening of ledger
+		if err := chaincodesLifecycleEventsProvider.RegisterListener(
+			l.ledgerID, &ccEventListenerAdaptor{stateDBIndexCreator}, false); err != nil {
+			return err
+		}
+		legacyChaincodesLifecycleEventsProvider.Register(l.ledgerID, stateDBIndexCreator)
+		return nil
+	}
+
+	// opening of ledger after creating from a snapshot -
+	// it would have been better if we could explicitly retrieve the list of invocable chaincodes instead of
+	// passing the flag initializer.initializingFromSnapshot to the ccLifecycleEventProvider (which is essentially
+	// the _lifecycle cache) for directing ccLifecycleEventProvider to call us back. However, the lock that ensures
+	// the synchronization with the chaincode installer is maintained in the lifecycle cache and by design the lifecycle
+	// cache takes the responsibility of calling any listerner under the lock
+	if err := chaincodesLifecycleEventsProvider.RegisterListener(
+		l.ledgerID, &ccEventListenerAdaptor{stateDBIndexCreator}, true); err != nil {
+		return errors.WithMessage(err, "error while creating statdb indexes after bootstrapping from snapshot")
```

Comment:
could probably pull `error while creating statdb indexes after bootstrapping from snapshot` into a constant

Reply:
Sure ... more detailed error is fine if you think it will provide better context than the underlying wrapped error.

Reply:
For now I don't think that more context is needed until going forward we realize the need. Just that binding these via a common constant gives an impression that we want to keep these same - which is not the intention.